JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -237,8 +249,12 @@ private async Task<int> GetNumberOfMessagesToReadFromSqs()
 
         private Task HandleMessage(Amazon.SQS.Model.Message message, CancellationToken ct)
         {
-            var action = new Func<Task>(() => _messageDispatcher.DispatchMessage(message, ct));
-            return _messageProcessingStrategy.StartWorker(action, ct);
+            async Task DispatchAsync()
+            {
+                await _messageDispatcher.DispatchMessage(message, ct);
```

Comment:
Minor - could elide the `await` here

Reply:
I figured to keep it to make debug more predictable (I’m sure there’s something in David Fowler’s async tips about that).

Reply:
[Found it](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task)

Reply:
For my curiosity, is this still an allocations saving? i.e. Does a `Task` allocate more than a `Func`? (my gut says yes).

Reply:
I did this more for readability after the changes I did in #549 where I needed to make the action do more work and then it felt a bit bloated and bleurgh for a lambda.